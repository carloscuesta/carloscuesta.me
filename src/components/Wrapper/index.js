// @flow
import React, { type Node } from 'react'

import styles from './styles.module.css'

type Props = { children: Node }

const Wrapper = (props: Props) => (
  <div className={styles.wrapper}>
    {props.children}
  </div>
)

export default Wrapper
