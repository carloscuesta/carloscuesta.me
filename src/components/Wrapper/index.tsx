type Props = {
  children: JSX.Element[] | JSX.Element
  isCompressed?: boolean
}

const Wrapper = (props: Props) => (
  <div
    className={`m-[0_auto] max-w-4xl px-6 sm:px-4 ${
      props.isCompressed ? 'max-w-[800px]' : ''
    }`}
  >
    {props.children}
  </div>
)

export default Wrapper
