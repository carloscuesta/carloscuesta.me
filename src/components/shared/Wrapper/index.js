// @flow
import type { Node, Element } from 'react'

import styles from './styles.module.css'

type Props = { children: Node, isCompressed?: boolean }

const Wrapper = (props: Props): Element<'div'> => (
  <div className={props.isCompressed ? styles.wrapperCompressed : styles.wrapper}>
    {props.children}
  </div>
)

export default Wrapper
