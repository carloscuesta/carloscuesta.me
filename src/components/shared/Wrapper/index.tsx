import styles from './styles.module.css'

type Props = { children: JSX.Element[] | JSX.Element, isCompressed?: boolean }

const Wrapper = (props: Props) => (
  <div className={props.isCompressed ? styles.wrapperCompressed : styles.wrapper}>
    {props.children}
  </div>
)

export default Wrapper
