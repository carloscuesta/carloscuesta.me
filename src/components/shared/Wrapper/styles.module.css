.wrapper {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 1.5em;
}

.wrapperCompressed {
  max-width: 824px;
  margin: 0 auto;
  padding: 0 1.5em;
}
