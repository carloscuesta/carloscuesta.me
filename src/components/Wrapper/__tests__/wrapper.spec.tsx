import renderer from 'react-test-renderer'
import Wrapper from '../index'

describe('Wrapper', () => {
  it('should render the component with children', () => {
    const wrapper = renderer.create(
      <Wrapper>
        <h1>Some children</h1>
        <h2>Hello!</h2>
      </Wrapper>,
    )

    expect(wrapper).toMatchSnapshot()
  })

  describe('isCompressed', () => {
    it('should render the component with children', () => {
      const wrapper = renderer.create(
        <Wrapper isCompressed>
          <h1>Some children</h1>
          <h2>Hello!</h2>
        </Wrapper>,
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
