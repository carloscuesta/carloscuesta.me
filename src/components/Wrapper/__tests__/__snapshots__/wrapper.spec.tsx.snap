// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper isCompressed should render the component with children 1`] = `
<div
  className="m-[0_auto] max-w-4xl px-6 sm:px-4 max-w-[800px]"
>
  <h1>
    Some children
  </h1>
  <h2>
    Hello!
  </h2>
</div>
`;

exports[`Wrapper should render the component with children 1`] = `
<div
  className="m-[0_auto] max-w-4xl px-6 sm:px-4 "
>
  <h1>
    Some children
  </h1>
  <h2>
    Hello!
  </h2>
</div>
`;
